src/backend/test.rs

```diff
@@ -102,6 +112,38 @@ impl TestBackend {
         crate::assert_buffer_eq!(&self.buffer, expected);
     }
 
+    /// Asserts that the `TestBackend`'s scrollback buffer is equal to the expected buffer.
+    ///
+    /// This is a shortcut for `assert_eq!(self.scrollback(), &expected)`.
+    ///
+    /// # Panics
+    /// When they are not equal, a panic occurs with a detailed error message showing the
+    /// differences between the expected and actual buffers.
+    #[allow(deprecated)]
+    #[track_caller]
+    pub fn assert_scrollback(&self, expected: &Buffer) {
+        // TODO: use assert_eq!()
+        crate::assert_buffer_eq!(&self.scrollback, expected);
+    }
+
+    /// Asserts that the `TestBackend`'s scrollback buffer is empty, but also has the given width.
+    ///
+    /// # Panics
+    /// When the scrollback buffer is not equal, a panic occurs with a detailed error message
+    /// showing the differences between the expected and actual buffers.
+    pub fn assert_empty_scrollback(&self) {
+        let expected = Buffer {
+            area: Rect {
+                x: 0,
+                y: 0,
+                width: self.scrollback.area.width,
+                height: 0,
+            },
+            content: vec![],
+        };
+        self.assert_scrollback(&expected);
+    }
```

Comment:
I have a change that renames the function as you suggested. There is already `assert_scrollback_lines`, so it only makes sense to have `assert_scrollback_empty`.

As for your first point, I'm not sure I follow about the implementation detail. This *is* the implementation, so I don't really see the issue with it knowing about the width. In the previous version of this function, it required the width to be passed in. @orhun pointed out that that was weird, and I agreed. But now the width detail is hidden in TestBackend implementaton.

I'm constructing a specific `Buffer` here so that `assert_scrollback` can be used, with all of its failure reporting goodness.